tests/Unit/IO/Test_H5.cpp

```diff
@@ -452,6 +459,106 @@ SPECTRE_TEST_CASE("Unit.IO.H5.DatRead", "[Unit][IO][H5]") {
   }
 }
 
+// Test that we can read scalar, rank-1, rank-2, and rank-3 datasets
+SPECTRE_TEST_CASE("Unit.IO.H5.ReadData", "[Unit][IO][H5]") {
+  const std::string h5_file_name("Unit.IO.H5.ReadData.h5");
+  const hid_t file_id =
+      H5Fcreate(h5_file_name.c_str(), H5F_ACC_TRUNC, H5P_DEFAULT, H5P_DEFAULT);
+  h5::detail::OpenGroup my_group(file_id, "ReadWrite",
+                                 h5::AccessType::ReadWrite);
+  const hid_t group_id = my_group.id();
+
+  using AllTypeList = tmpl::list<double, int, unsigned int, long, unsigned long,
+                                 long long, unsigned long long>;
+
+  const DataVector scalar_data{1.0};
```

Comment:
You should write more than one element, typically we do 5 elements for DataVectors

Reply:
Sorry, here you're writing a scalar, never mind.